### DISCUSSION TITLE
doxygen rule vs using make rule to call doxygen with Doxyfile

### DISCUSSION THREAD

**Discussion by mapiri** [bfgroup/b2#185] [General]
Hi
As I understand, the doxygen rule doesn't support the specification of a configuration file (Doxyfile), instead it expects to set all supported config parameters in the using statement with doxygen:param statements, which are then added to a Doxyfile that is generated by b2. Is this correct? If so, using this rule doesn't look very attractive compared with simply using a make rule and call doxygen directly with a predefined Doxyfile.
I am wondering whether there are any other reasons for using the doxygen rule instead of the make rule approach that I am overlooking.
Please let me know.

Thanks

**Comment by grisumbras** [comment]
The main benefit is that the C++ sources for doxygen invocation are collected from the sources of the target. If any source is changed or added, doxyfile is updated and doxygen re-run.

**Comment by mapiri** [comment]
Thanks for the prompt feedback. That's valuable information.
